stan/math/rev/fun/multiply_log.hpp

```diff
@@ -105,6 +105,265 @@ inline var multiply_log(double a, const var& b) {
+                                      arena_b.coeff(i, j)))) {
+                arena_a.adj().coeffRef(i, j) = NOT_A_NUMBER;
+              } else {
+                arena_a.adj().coeffRef(i, j)
+                    += res.adj().coeff(i, j) * log(arena_b.coeff(i, j));
+              }
+            }
+          }
+        });
+  } else {
+    auto arena_a = to_arena(value_of(a));
+    arena_t<promote_scalar_t<var, T2>> arena_b = b;
+
+    return make_callback_var(
+        multiply_log(arena_a, arena_b.val()),
+        [arena_a, arena_b](const auto& res) mutable {
+          for (Eigen::Index j = 0; j < res.adj().cols(); ++j) {
+            for (Eigen::Index i = 0; i < res.adj().rows(); ++i) {
+              if (unlikely(is_any_nan(arena_a.val().coeff(i, j),
+                                      arena_b.val().coeff(i, j)))) {
+                arena_b.adj().coeffRef(i, j) = NOT_A_NUMBER;
+              } else {
+                if (arena_b.val().coeff(i, j) == 0.0
+                    && arena_a.val().coeff(i, j) == 0) {
+                  arena_b.adj().coeffRef(i, j) += res.adj().coeff(i, j) * INFTY;
+                } else {
+                  arena_b.adj().coeffRef(i, j) += res.adj().coeff(i, j)
+                                                  * arena_a.coeff(i, j)
+                                                  / arena_b.val().coeff(i, j);
+                }
+              }
+            }
+          }
+        });
+  }
+}
+
+/**
+ * Return the product `a * log(b)`.
+ *
+ * @tparam T1 Type of matrix argument
+ * @tparam T2 Type of scalar argument
+ * @param a Matrix argument
+ * @param b Scalar argument
+ * @return Product of `a` and `log(b)`
+ */
+template <typename T1, typename T2, require_var_matrix_t<T1>* = nullptr,
+          require_stan_scalar_t<T2>* = nullptr>
+inline auto multiply_log(const T1& a, const T2& b) {
+  using std::log;
+
+  if (!is_constant<T1>::value && !is_constant<T2>::value) {
+    arena_t<promote_scalar_t<var, T1>> arena_a = a;
+    var arena_b = b;
+
+    return make_callback_var(
+        multiply_log(arena_a.val(), arena_b.val()),
+        [arena_a, arena_b](const auto& res) mutable {
+          for (Eigen::Index j = 0; j < res.adj().cols(); ++j) {
+            for (Eigen::Index i = 0; i < res.adj().rows(); ++i) {
+              if (unlikely(
+                      is_any_nan(arena_a.val().coeff(i, j), arena_b.val()))) {
+                arena_a.adj().coeffRef(i, j) = NOT_A_NUMBER;
+                arena_b.adj() = NOT_A_NUMBER;
+              } else {
+                arena_a.adj().coeffRef(i, j)
+                    += res.adj().coeff(i, j) * log(arena_b.val());
+                if (arena_b.val() == 0.0 && arena_a.val().coeff(i, j) == 0) {
+                  arena_b.adj() += res.adj().coeff(i, j) * INFTY;
+                } else {
+                  arena_b.adj() += res.adj().coeff(i, j)
+                                   * arena_a.val().coeff(i, j) / arena_b.val();
```

Comment:
If your using `res.adj().coeff(i,j)` in a bunch of places it can be a little nicer asm if you pull it out